### PR TITLE
Fixed validation issues on add task and add requirement.

### PR DESCRIPTION
**Pull request by D-B-Cooper**
Fixed issue #313 which allowed the ok button to be pressed despite having the invalid user input. This applied to both add task and add requirement.

### CODE REVIEW

File: src/edu/wright/cs/raiderplanner/controller/TaskController.java [hunk 1]
```diff
@@ -164,7 +157,31 @@ public void handleChange() {
 		}
 		// =================
 	}
-
+	
+	/**
+	 * Used to test all user entries needed before allowing the ok/submit button to be
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/TaskController.java]: This comment needs to be a bit more descriptive.

File: src/edu/wright/cs/raiderplanner/controller/TaskController.java [hunk 2]
```diff
@@ -164,7 +157,31 @@ public void handleChange() {
...
 	}
-
+	
+	/**
+	 * Used to test all user entries needed before allowing the ok/submit button to be 
+	 * 	pressed.  
+	 * @return true if unlock is successful, or false if not
+	 */
+	public boolean unlockSubmit() {
+		if (!this.name.getText().trim().isEmpty()
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/TaskController.java]: Are we sure that the text field contents will always be non-null?  Perhaps we should something like `this.name.getText() != null && this.name.getText.trim().isEmpty() ....`

File: src/edu/wright/cs/raiderplanner/controller/TaskController.java [hunk 3]
```diff
@@ -132,18 +132,11 @@ public boolean isSuccess() {
 
 	/**
 	 * Handle changes to the input fields.
+	 * @return
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/TaskController.java]: This looks to be extraneous.  Please remove.

File: src/edu/wright/cs/raiderplanner/controller/TaskController.java [hunk 4]
```diff
@@ -241,6 +258,7 @@ public void toggleComplete() {
 			this.completed.setVisible(true);
 			this.canComplete.setVisible(false);
 		}
+		this.unlockSubmit();
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/TaskController.java]: This method returns a boolean.  Why is it being ignored here?

Comment (D-B-Cooper) [reply src/edu/wright/cs/raiderplanner/controller/TaskController.java]: Whether it passes or fails is irrelevant. This is used to fix an issue where if a task is being edited, toggling the complete would not unlock the ok button. This is confusing for the user.

**Comment by D-B-Cooper** [comment]
The only errors I see when running checks are "trailing white space" errors on lines where it isn't applicable. Not sure how to fix.

**Pull request closed without merging**